Criando as tabelas e a conexão com o banco de dados

JeFFsPinh opened the pull request:
Foram criadas as tabelas seguindo os requisitos do projeto, e a conexão com o banco de dados utilizado no projeto foi estabelecida.

Commit:
Adicionando os models das tabelas e criando conexao com o banco.

Foram criados os models cumprindo os requisitos de cada tabela e foi feita a configuraçaoo para conexao com o banco de dados.

JeFFsPinh committed:
Update main.py

igormcsouza commented on app.py:
```diff
@@ -1,7 +1,12 @@
 from flask import Flask
+from project.infrastructure.database import main
+from flask_migrate import Migrate
```
Eu prefiro fazer no braço mas pode ser! :D

igormcsouza commented on app.py:
```diff
@@ -1,7 +1,12 @@
 from flask import Flask
+from project.infrastructure.database import main
```
Dentro do `../database/__init__.py` você colocar um import para o init_app que está no main.py e aqui você importa o database e faz database.init_app()

igormcsouza commented on docker-compose.yml:
```diff
@@ -2,18 +2,24 @@ version: '3.8'
+    volumes:
+      - ./data/postgres:/var/lib/postgresql/data
     ports:
       - 5432:5432
```
Adicionar nova linha ao fnal do arquivo

igormcsouza commented on project/infrastructure/database/account/model.py:
```diff
@@ -0,0 +1,19 @@
+            "name": self.name,
+            "starting_balance": self.starting_balance,
+            "id_user": self.id_user
+        }
```
Nova linha ao final de todos os arquivos!

igormcsouza commented on project/infrastructure/database/balance/model.py:
```diff
@@ -0,0 +1,18 @@
+from sqlalchemy import Column, ForeignKey, Integer
+
+
+class Account(Base):
```
Nome da classe deve ser Balance

igormcsouza commented on project/infrastructure/database/installments/model.py:
```diff
@@ -0,0 +1,20 @@
+from sqlalchemy import Column, Integer, Boolean, Date
+
+
+class Account(Base):
```
Nome da classe deve ser Installments

igormcsouza commented on project/infrastructure/database/kind/model.py:
```diff
@@ -0,0 +1,15 @@
+from sqlalchemy import Column, Integer, Boolean, Date, String
+
+
+class Account(Base):
```
Nome da classe deve ser Kind

igormcsouza commented on project/infrastructure/database/kind/model.py:
```diff
@@ -0,0 +1,15 @@
+    __tablebame__ = 'Kind'
+
+    id = Column(Integer, primary_key = True)
+    name = Column(String[70], nullable = False)
```
É só para setar entrada e saída.. pode ser uma string de 10 posições só!

igormcsouza commented on project/infrastructure/database/main.py:
```diff
@@ -0,0 +1,17 @@
+Session = sessionmaker(bind = engine)
+Base = declarative_base()
+
+def init_app():
```
Dessa forma não precisa de migrate

igormcsouza commented on project/infrastructure/database/transactions/model.py:
```diff
@@ -0,0 +1,30 @@
+from sqlalchemy import Column, Float, ForeignKey, Integer, Boolean, Date, String
+
+
+class Account(Base):
```
Nome da classe deve ser Transaction

igormcsouza commented on project/infrastructure/database/user/model.py:
```diff
@@ -0,0 +1,20 @@
+            "id": self.id,
+            "username": self.username,
+            "email": self.email,
+            "password": self.password
```
Nunca inclua o password de um usuário em uma estrutura que pode ser enviada no response... Para não deixar o password cair em mãos erradas

igormcsouza reviewed (changes requested):
Tem apenas algumas besteirinhas para ver.. Mas queria conversar com vocês sobre migrations!

Commit:
Adicionando migrations ao projeto e corrigindo estrutura de arquivos

Foi utilizado o alembic para criar as migrations aplicadas no projeto e foram corrigidos alguns arquivos que possuiam erros em sua estrutura, aplicando a convençao dos padroes adotados pelo python para o codigo

Commit:
Atualizaçao do arquivo para conexao ao banco

igormcsouza committed:
Update .env

igormcsouza committed:
Update docker-compose.yml

igormcsouza commented on alembic.ini:
```diff
@@ -0,0 +1,100 @@
+# are written from script.py.mako
+# output_encoding = utf-8
+
+sqlalchemy.url = unknown
```
Faltou a URI do postgres para se conectar

igormcsouza replied:
Notei que no env.py vc carregou as variaveis de ambiente! show!

igormcsouza committed:
Update model.py

igormcsouza committed:
Update model.py

igormcsouza committed:
Update model.py

igormcsouza committed:
Update model.py

igormcsouza committed:
Update model.py

igormcsouza reviewed (changes requested):
Está excelente! Faltou só mesmo o uri do postgres! :D

igormcsouza reviewed (approved):
Show!! Tá ótimo!
